iphone/cli/lib/process-js-task.js

```diff
@@ -0,0 +1,374 @@
+'use strict';
+
+const appc = require('node-appc');
+const { IncrementalFileTask } = require('appc-tasks');
+const crypto = require('crypto');
+const fs = require('fs-extra');
+const jsanalyze = require('node-titanium-sdk/lib/jsanalyze');
+const path = require('path');
+
+const i18n = appc.i18n(__dirname);
+const __ = i18n.__;
+
+/**
+ * Task that processes JS files by applying several transforms and copying them to their
+ * destination inside the build directory.
+ */
+class ProcessJsTask extends IncrementalFileTask {
+	/**
+	 * Constructs a new processing task.
+	 *
+	 * @param {Object} taskInfo Configuration object for this task
```

Comment:
please document the optional/required properties on the taskInfo object

```suggestion
	 * @param {object} taskInfo Configuration object for this task
	 * @param {string} [taskInfo.name='process-js'] name for the task
	 * @param {string[]} taskInfo.jsFiles array of file paths to consider
	 * @param {string[]} taskInfo.jsBootstrapFiles array of bootstrap scripts to consider
	 * @param {object} taskInfo.builder the iOS Builder object driving the CLI build
	 * @param {boolean} taskInfo.builder.minifyJS whether to minify the JS files
	 * @param {boolean} taskInfo.builder.transpile whether to transpile the JS files
	 * @param {boolean} taskInfo.builder.sourceMaps whether to generate source maps
	 * @param {string} taskInfo.builder.deployType 'production' or 'development'
	 * @param {string} taskInfo.builder.xcodeAppDir path to app directory
	 * @param {string} taskInfo.builder.minSupportedIosSdk minimum iOS version to support in transpiling
	 * @param {string} taskInfo.sdkCommonFolder path to the SDK's common js API folder
```